explainability/explainability-core/src/main/java/org/kie/kogito/explainability/model/NumericFeatureDistribution.java

```
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.kogito.explainability.model;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Numeric feature distribution based on {@code double[]}.
+ */
+public class NumericFeatureDistribution implements FeatureDistribution {
+
+    private final Logger LOGGER = LoggerFactory.getLogger(NumericFeatureDistribution.class);
+
+    private final Feature feature;
+    private final double[] doubles;
+
+    public NumericFeatureDistribution(Feature feature, double[] doubles) {
+        this.feature = feature;
+        this.doubles = doubles;
+    }
+
+    @Override
+    public Feature getFeature() {
+        return feature;
+    }
+
+    @Override
+    public Value<?> sample() {
+        return sample(1).get(0);
+    }
+
+    @Override
+    public List<Value<?>> sample(int sampleSize) {
+        if (sampleSize >= doubles.length) {
+            LOGGER.warn("required {} samples, but only {} are available", sampleSize, doubles.length);
```

Shouldn't we rather throw an exception or something. User may not notice this in logs.

---

I've made it possible to sample a no. of samples bigger than the sample size (so generated data will contain duplicates).